dyno-core/src/main/java/com/netflix/dyno/connectionpool/Host.java

```diff
@@ -122,13 +122,18 @@ public InetSocketAddress getSocketAddress() {
         return socketAddress;
     }
 
+    /**
+     * The HostSupplier may not have the ability to return the port
+     * and therefore the hashCode() may fail if the user provides his
+     * own port. Hence, we do not have to include the port number in the
+     * hashCode() calculation.
+     */
     @Override
     public int hashCode() {
         final int prime = 31;
         int result = 1;
         result = prime * result + ((hostname == null) ? 0 : hostname.hashCode());
         result = prime * result + ((rack == null) ? 0 : rack.hashCode());
```

Comment:
We do not and even if we wanted the whole process would be fairly complicated to implement. Eureka provides a set of IP addresses and these are cross-validated with Florida's `cluster_describe` hostnames/tokens. Every hostname maps to a token, hence we have to change a lot of pieces to make a hostname mapping multiple tokens.